### DISCUSSION TITLE
Try out podman 3.0 and tell us what you think?

### DISCUSSION THREAD

**Discussion by rhatdan** [containers/podman#9059] [General]
We really need people testing and playing with this. We want to fix as many issues/bugs as possible.  Especially play with the remote API.  

Run docker-py and docker-compose against.  Try to break it.

**Comment by Hendrik-H** [comment]
Hi, as you are talking about special interest in bugs regarding the remote API. Any chance somebody could take a look at https://github.com/containers/podman/issues/8713 ? ;-)

**Reply by PavelSosin-320** [reply]
Currently, I'm running Podman 3.0 on the Windows Pro (insider release), WSL2 Fedora33 (Unofficial distro) with systemd activated using Arkane-system's systemd-genie.. I'm testing some small Pod built around old Erich Gamma idea how to unify the architecture of 3 EDI's: Microsoft VSCode, RedHat Eclispe/Che and SAP WebIDE. The Cloud IDE backend is a Pod containing Workspace manager and side-car extension containers: Editor's backend like Theia, WebIDEElectron. Plus plugind: LSP, DAP, Programming language packs extensions. All Containers share the Workspace volume and communicate via a network. Meanwhile, Podman satisfies such architecture requirements. Cons: a lot of adjustments in the distro, WSL, and genie, not in the code but mainly how to use all these building blocks together.

**Comment by trusch** [comment]
I just tried out if network aliases are now working in rootless networks, but I got disappointed. The issue is stale for nearly two weeks. This supprises me, how does all the docker-compose stuff work if containers can only reach each another using their real container-names?

**Reply by Luap99** [reply]
@trusch You can expect it in the 3.0 release once https://github.com/containers/podman/pull/9141 is merged.

**Reply by trusch** [reply]
Sorry for the issue from yesterday regarding this, that was obviously my fault there. I love to see this working now :) I just tried it out with an updated podman version and it worked like a charm. Very nice work! @Luap99

**Comment by rhatdan** [comment]
@baude ^^

**Comment by siretart** [comment]
I've uploaded 3.0rc2 to [debian/experimental](https://packages.qa.debian.org/libp/libpod/news/20210131T130340Z.html) and tried it with the debian packaged docker-compose following the instructions from https://www.redhat.com/sysadmin/podman-docker-compose -- works fine for me on a laptop.

I've tried it (first) on rootless, but that doesn't work (as advertised/explained in the link above):

```
siretart@x1:~/git/awesome-compose/gitea-postgres$ docker-compose -H unix:/run/user/1000/podman/podman.sock up
Pulling gitea (gitea/gitea:latest)...
76146dacacd305f5bbb4af41b6d107c06679db432fd1b5de4dd9577a56789dcf: pulling image () from docker.io/gitea/gitea:latest 
Pulling db (postgres:alpine)...
8c6053d81a45b455a79e7abbc5b4d8410735f060634d345f70b0530b1d50b2b6: pulling image () from docker.io/library/postgres:alpine 
Creating gitea-postgres_gitea_1 ... error
Creating gitea-postgres_db_1    ... 

Creating gitea-postgres_db_1    ... error

ERROR: for gitea-postgres_db_1  network connect is not enabled for rootless containers

ERROR: for gitea  network connect is not enabled for rootless containers

ERROR: for db  network connect is not enabled for rootless containers
ERROR: Encountered errors while bringing up the project.
```

What would it take to make network connect work for rootless mode? Is that on the roadmap?

**Reply by vrothberg** [reply]
Rootless docker-compose is not yet supported yet. @mheon and @baude were brainstorming this week how to enable that in the future.

**Comment by mheon** [comment]
Definitely doable, but no one on the team has taken it up yet. Probably not
that difficult. Might be worth a GitHub enhancement request?

**Reply by siretart** [reply]
Good idea, filed #9169 for that.

**Comment by d3xter** [comment]
I've tried to give it a whirl on fedora 33 by using `baude/Upstream_CRIO_Family` COPR repo, but there is only a `podman` package and no `podman-docker`. Where can I get this package or do I have to do it manually/use the podman socket?

**Reply by mheon** [reply]
I think you'll have to do it manually, but all you need to do is place a copy of https://github.com/containers/podman/blob/master/docker somewhere in your $PATH

**Comment by tanguydelignieresaccenture** [comment]
Hi, I'm trying podman 3.0 with a "PiD" approach  (podman in docker with a UBI 8 docker container, crun and cgroups v1) from docker desktop on Windows for local development objectives with podman.

I was able to run rootfull containers stack via docker-compose (v2, no volume) directly from the docker container without `--privileged` flag. So far so good, that's a really good achievement after having faced quite a lot of issues (docker, podman, ubi, ...).

Now I'm stuck in trying podman from my host with podman remote as I cannot find a pre-release of the windows zip/msi. Is it available somewhere or do I need to build it?
If building is required how to be sure to build the correct version (as podman info does not gives me commit information, only version > 3.0.0-rc1)?

**Reply by vrothberg** [reply]
We don't yet provide builds of pre-releases.  You can build it by downloading the source code and doing `make podman-remote`.

**Comment by patrikwm** [comment]
Hi, Im using `3.2.0-rc3` and trying to figure out how to use `--net=slirp4netns:port_handler=slirp4netns` in the docker-compose file. I need the source IP in the container. Seems that rootlesskit always sets source ip to 10.0.2.100.

**Reply by Luap99** [reply]
see https://github.com/containers/podman/issues/10110

**Reply by patrikwm** [reply]
Thanks! that worked, but now i lost the connection to the database service that is in the same pod (docker-compose) file. Any tips and tricks for this?

I just notice that the host is not listening for the specified ports anymore either. so following does not work anymore.
```
    ports:
      - "8443:8443/tcp"
```

**Reply by patrikwm** [reply]
With following docker-compose file 
```
version: '3'
services:
  netcat:
    image: busybox
    command: nc -lv -p 80
    ports:
      - 8080:80/tcp
    network_mode: "slirp4netns:port_handler=slirp4netns"
```
i do following
```
podman-compose up -d

using podman version: podman version 3.2.0-rc3
podman pod create --name=netcat --share net -p 8080:80/tcp
552f6d8a2fd0cc1c7f5240aaa727293f3993b5c6318e1dab455e431f89a85671
0
podman run --name=netcat_netcat_1 -d --pod=netcat --label io.podman.compose.config-hash=123 --label io.podman.compose.project=netcat --label io.podman.compose.version=0.0.1 --label com.docker.compose.container-number=1 --label com.docker.compose.service=netcat --network slirp4netns:port_handler=slirp4netns --add-host netcat:127.0.0.1 --add-host netcat_netcat_1:127.0.0.1 busybox nc -lv -p 80
5a8896cdefcf30d01294ff4d234c17c8ee84d85995408bcc1d8c75e4c952bec7
0
```
and when i run netstat there is no port listening on port 8080.
```
sudo netstat -tnlp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN      1/systemd           
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      667/sshd            
tcp6       0      0 :::111                  :::*                    LISTEN      1/systemd           
tcp6       0      0 :::22                   :::*                    LISTEN      667/sshd 
```

**Reply by Luap99** [reply]
podman-compose is doing more stuff, the `--network slirp4netns:port_handler=slirp4netns` must be set on the podman pod create command. I recommend asking there how you can set this for the pod.
